.cascade-code/Chapman.edu/_components/omni_nav_v2/omni_nav.vtl

```diff
@@ -0,0 +1,812 @@
+          <ul>
+            <li><a href="/about/maps-directions/campus-map/">Interactive Map</a></li>
+            <li><a href="/our-faculty/">Faculty Directory</a></li>
+            <li><a href="/directory/">All Directories</a></li>
+          </ul>
+        </li>
+        <li>
+          <a href="https://blogs.chapman.edu/">Blogs</a>
+          #buildToggleIcon()
+          <ul>
+            <li><a href="https://blogs.chapman.edu/happenings/">Happenings</a></li>
+            <li><a href="https://blogs.chapman.edu/alumni/">Chapman Alumni</a></li>
+            <li><a href="https://blogs.chapman.edu/business/">Argyros School of Business &amp; Economics</a></li>
+            <li><a href="https://blogs.chapman.edu/education/">Attallah College of Educational Studies</a></li>
+            <li><a href="https://blogs.chapman.edu/copa/">College of Performing Arts</a></li>
+            <li><a href="https://blogs.chapman.edu/crean/">Crean College of Health &amp; Behavioral Sciences</a></li>
+            <li><a href="https://blogs.chapman.edu/dodge/">Dodge College of Film &amp; Media Arts</a></li>
+            <li><a href="https://blogs.chapman.edu/law/">Fowler School of Law</a></li>
+            <li><a href="https://blogs.chapman.edu/scst/">Schmid College of Science &amp; Technology</a></li>
+            <li><a href="https://blogs.chapman.edu/communication/">School of Communication</a></li>
+            <li><a href="https://blogs.chapman.edu/pharmacy/">School of Pharmacy</a></li>
+            <li><a href="https://blogs.chapman.edu/wilkinson/">Wilkinson College of Arts, Humanities, &amp; Social Sciences</a></li>
+            <li><a href="https://blogs.chapman.edu/">View More Blogs</a></li>
+          </ul>
+        </li>
+        <li><a href="https://events.chapman.edu/">Events</a></li>
+        <li><a href="https://social.chapman.edu/">Social</a></li>
+      </ul>
+    </div>
+  </div>
+#end
+
+#macro ( buildOffCanvasNavSecondaryMenu )
+  <div id="off-canvas-umbrella" class="off-canvas-menu">
+    <div class="menu-header">
+      <span class="menu-label">${umbrellaLabel}</span>
+      <a class="toggle-menu-label" href="#">Main Menu &gt;</a>
+    </div>
+
+    <ul>
+      #foreach ( $asset in $umbrellaAssets )
+        #buildSecondaryMenuItem( $asset )
+      #end
+    </ul>
+  </div>
+#end
+
+#macro ( buildSecondaryMenuItem $asset )
+  ## This builds a menu link for the secondary menu. It will either be a link or a
+  ## drop-down menu with multiple links.
+  #set ( $metaHideSetting = $asset.metadata.getDynamicField('Hide from navigation').value )
+  #set ( $isVisible = $metaHideSetting != 'Yes' )
+  #set ( $assetType = $asset.assetType )
+
+  ## If hidden, continue. But velocity doesn't have a continue operator:
+  ## https://stackoverflow.com/questions/7530019/apache-velocity-foreach-loop-continue
+  #if ( $isVisible )
+    #if ( $assetType == 'page' )
+      <li><a href="${asset.path}">${_EscapeTool.xml($asset.metadata.displayName)}</a></li>
```

Comment:
I would not expect `_EscapeTool` to be required here. But it's hard to tell with the Adapter API methods. And error messages when it is needed are not clear.